sw/airborne/firmwares/rotorcraft/guidance/guidance_v.c

```diff
@@ -481,8 +481,18 @@ static void run_hover_loop(bool in_flight)
 bool guidance_v_set_guided_z(float z)
 {
   if (guidance_v_mode == GUIDANCE_V_MODE_GUIDED) {
+    /* disable vertical velocity setpoints */
     guidance_v_guided_vel_enabled = false;
+
+    /* set altitude setpoint */
     guidance_v_z_sp = POS_BFP_OF_REAL(z);
+
+    /* reset speed setting */
+    guidance_v_zd_sp = 0;
+
+    /* reset guidance reference */
+    guidance_v_z_sum_err = 0;
+    GuidanceVSetRef(guidance_v_z_sp, guidance_v_zd_sp, 0);
```

Comment:
why do you want to reset the ref every time you change the setpoint?
IMHO that is not what you usually want... you might want to reset the ref on entering the guided mode, but not on every update...